SoftwareGuide/Latex/Appendices/CodingStyleGuide.tex

```diff
@@ -1428,6 +1430,29 @@ \section{Pipelines}
 
 \end{itemize}
 
+\section{The auto Keyword}
+\label{sec:TheAutoKeyword}
+
+Available since C++11, the \code{auto} keyword specifies that a variable's type
+is automatically deduced from its initializer.
+
+The \code{auto} keyword should be used to specify a type in the following cases:
+
+\begin{itemize}
+\item The type is duplicated on the left side of an assigment when it is mandated on the
+      right side, e.g. when there is an explicit cast or initializing with \code{new}.
+\item When obtaining container elements, when the element type is obvious from the type
+      of the container.
+\item When declaring iterators in a \code{for} loop.
```

Comment:
Why not add VTK's "When the type doesn't matter because it isn't being used for anything other than equality comparison."

Reply:
If this is added, I wouldn't use contractions. I would keep the language rather formal in the SWG.

Reply:
Will add without the contractions.